out_writeapi_test.go

```diff
@@ -38,3 +47,244 @@ func TestFLBPluginRegister(t *testing.T) {
+						{Name: "Text", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+					},
+				},
+			}, nil
+		},
+	}
+
+	originalFunc := getClient
+	getClient = func(ctx context.Context, projectID string) (ManagedWriterClient, error) {
+		log.Println("Mock ManagedWriterClient called")
+		return mockClient, nil
+	}
+	defer func() { getClient = originalFunc }()
+
+	res := []bool{}
+	md, _ := getDescriptors(ms_ctx, mockClient, "dummy", "dummy", "dummy")
+	mockMS := &MockManagedStream{
+		AppendRowsFunc: func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+			checks.appendRows++
+
+			var combinedData []byte
+			for _, tempData := range data {
+				combinedData = append(combinedData, tempData...)
+			}
+
+			message := dynamicpb.NewMessage(md)
+			err := proto.Unmarshal(combinedData, message)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to unmarshal")
+			}
+
+			textField := message.Get(md.Fields().ByJSONName("Text"))
+			timeField := message.Get(md.Fields().ByJSONName("Time"))
+
+			if textField.String() != "FOO" && textField.String() != "BAR" {
+				return nil, fmt.Errorf("Wrong Text Value: %s", textField.String())
+			}
+			log.Printf("String: %s", textField.String())
+			if timeField.String() != "000" {
+				return nil, fmt.Errorf("Wrong Time Value")
+			}
+
+			res = append(res, true)
+			res = append(res, true)
+			return nil, nil
+		},
+		FinalizeFunc: func(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		FlushRowsFunc: func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		StreamNameFunc: func() string {
+			return ""
+		},
+	}
+
+	origFunc := getWriter
+	getWriter = func(client ManagedWriterClient, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = origFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		return ""
+	})
+	defer patch1.Unpatch()
+
+	patchSetContext := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		setID = ctx.(int)
+	})
+	defer patchSetContext.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	initRes := FLBPluginInit(plugin)
+
+	orgFunc := getFLBPluginContext
+	getFLBPluginContext = func(ctx unsafe.Pointer) int {
+		checks.calledGetContext++
+		return setID
+	}
+	defer func() { getFLBPluginContext = orgFunc }()
+
+	origReadyFunc := isReady
+	isReady = func(queueHead *managedwriter.AppendResult) bool {
+		if res[0] {
+			return true
+		} else {
+			return false
+		}
+	}
+	defer func() { isReady = origReadyFunc }()
+
+	origResultFunc := pluginGetResult
+	pluginGetResult = func(queueHead *managedwriter.AppendResult, ctx context.Context) (int64, error) {
```

Comment:
Here also you could simply always return -1, nil.

Why would you want to return a failure? Is there some scenario where you want to simulate a failure?

Reply:
I was hoping that if in the future we do want to simulate a failure the infrastructure is there, but I can remove it!